packages/pigeon/lib/generator_tools.dart

```diff
@@ -432,3 +432,39 @@ Iterable<EnumeratedClass> getCodecClasses(Api api, Root root) sync* {
 /// Returns true if the [TypeDeclaration] represents an enum.
 bool isEnum(Root root, TypeDeclaration type) =>
     root.enums.map((Enum e) => e.name).contains(type.baseName);
+
+/// Formats documentation comments and adds them to current Indent.
+void addDocumentationComments(Indent indent, List<String> comments, String open,
+    {String? close, List<String>? additionalComments}) {
+  if (comments.isEmpty && additionalComments == null) {
+    return;
+  }
+  if (comments.length > 1 ||
+      (comments.isNotEmpty && additionalComments != null) ||
+      (additionalComments != null && additionalComments.length > 1)) {
+    bool addSymbolOnNextLine = true;
+    if (close != null) {
+      indent.writeln(open);
+      addSymbolOnNextLine = false;
+    }
+    if (additionalComments != null) {
+      for (final String line in additionalComments) {
+        indent.writeln(
+          '${addSymbolOnNextLine ? '$open ' : ''}$line',
+        );
+      }
+    }
+    for (final String line in comments) {
+      indent.writeln(
+        '${addSymbolOnNextLine ? '$open ' : ''}$line',
+      );
+    }
+    if (close != null) {
+      indent.writeln(close);
+    }
+  } else {
+    indent.writeln(
+      '$open ${comments.isNotEmpty ? comments.first : ''}${additionalComments?.first ?? ''} ${close ?? ''}',
+    );
+  }
```

Comment:
In fact, I think you could probably simplify all of this logic quite a bit if you start with:

```
List<String> allComments = <String>[
  ...comments,
  if (comments.isNotEmpty && additionalComments.isNotEmpty) '',
  ...additionalComments,
];
```

and then operate only on `allComments`.